<img style="float: left; margin: 30px 15px 15px 15px;" src="https://oci02.img.iteso.mx/Identidades-De-Instancia/ITESO/Logos%20ITESO/Logo-ITESO-Principal.jpg" width="400" height="600" /> 
    
    
##  Portafolios de Inversión
    
###   Growth Investing

Profesor: Sean Nicolás González Vázquez

###  Análisis Fundamental  
    
Tipo de análisis financiero que, como su nombre indica, se basa en los *fundamentales* de una empresa, es decir, en la información presentada en los tres principales estados financieros (Estado de Resultados, Flujo de Efectivo y Balance General) y en ratios calculados sobre la información presentada en estos tres estados financieros.  
    
El objetivo principal del análisis fundamental es determinar la salud financiera de una empresa, con el objetivo de ver si es conveniente invertir o no en esta, lo cual se decide en función de la metodología de inversión del gestor de inversiones.
    
El análisis fundamental es una herramienta que se utiliza para fundamentar la selección de un activo. La selección de activos es el primer paso del proceso de inversión en un portafolio.

###  Value Investing   
    

El **Value Investing**, es una fílosofia de inversión que busca invertir en negocios que tengan un valor de mercado menor a su valor intrínseco con la expectativa de que el mercado las revalúe correctamente. En término simples, esto significa comprar barato y vender caro. 
    

El Value Investing busca que el **valor extrínseco de una emisora** sea menor al **valor intrínseco dado a la emisora**, para buscar posibles **inflavaloraciones** o **sobrevaloraciones** de mercado:
    
+ *Acciones infravaloradas*: compañías que cotizan por debajo de su valor intrínseco, es decir, están baratas y sería conveniente comprar.  
    

+ *Acciones sobrevaloradas*: compañías que cotizan por encima de su valor intrínseco. Están caras y no sería buena opción de compra.
    
    
El concepto se mejoró posteriormente introduciendo el **long term value investing**, donde se buscan acciones que coticen a descuento, o cerca al *fair price* (Book Value = Market Value) y además, tengan una "*ventaja competitiva con durabilidad*".
    
 
###  Growth Investing   
    
    
El método inversión en *crecimiento* es una **estrategia que se centra en comprar acciones de empresas en fase de expansión cuyo valor se espera que crezca a tasas superiores al promedio del mercado**. En otras palabras, se buscan empresas con un potencial significativo de expansión en ingresos y ganancias en el futuro, además de una ventaja competitiva. Esta estrategia se basa en la creencia de que el crecimiento sostenido de una empresa impulsará el aumento del precio de sus acciones a largo plazo.

Algunas características clave del Growth Investing:

+ **Enfocado en el Crecimiento Empresarial**: Los inversores buscan empresas con sólidos fundamentos y un historial de crecimiento en ventas, beneficios y market share. 
    
    
+ **Poca Importancia a los Dividendos**: A diferencia de la estrategia de inversión en valor, donde los inversores buscan acciones subvaluadas con dividendos estables, la fílosofia de growth investing no prioriza los dividendos y se buscan empresas que reinvierten sus beneficios para financiar el crecimiento.  
    
    
+ **Alto Potencial de Rentabilidad a Largo Plazo**: La premisa básica del Growth Investing es que las empresas con un fuerte crecimiento tienen el potencial de generar rendimientos significativos a largo plazo, independientemente del valor actual. Estos inversores buscan capturar la apreciación del valor de las acciones a medida que la empresa crece.
    
    
+ **Volatilidad y Riesgo**: Dado que las empresas de crecimiento a menudo operan en industrias dinámicas y emergentes, las acciones de estas empresas pueden experimentar una mayor volatilidad en comparación con las empresas más establecidas. Existe un riesgo asociado con la incertidumbre sobre la capacidad de mantener altas tasas de crecimiento.
    
    
+ **Evaluación de Métricas de Crecimiento**: Los inversores de crecimiento analizan métricas clave como las tasas de crecimiento de ingresos, ganancias y márgenes para evaluar el potencial de crecimiento de una empresa. 
     

    
Ejemplos de empresas que podrían atraer a inversores de crecimiento incluyen aquellas en sectores tecnológicos, inteligencia artificial, biotecnología y otras industrias en rápido desarrollo.

Es importante destacar que el Growth Investing implica un enfoque a largo plazo y requiere paciencia, ya que el crecimiento de las empresas puede llevar tiempo en materializarse. Además, la selección cuidadosa de empresas con fundamentos sólidos es esencial para mitigar riesgos en esta estrategia.
    
+ **Importante**: Al invertir en empresas con una alta expectativa de crecimiento en el futuro, los inversionistas esperan también un retorno alto, por tanto, hay que estar conscientes que la volatilidad será mucho mayor que invertir en empresas consolidadas.
    
    
###  Diferencias Growth y Value  

    
| **Características**        | **Value Investing**                                                                 | **Growth Investing**                                                       |
|:---------------------------:|:--------------------------------------------------------------------------------------:|:----------------------------------------------------------------------------------:|
| **Enfoque Principal**       | Busca acciones subvaloradas o en descuento.                                           | Busca acciones de empresas con alto potencial de crecimiento.                     |
| **Criterio de Selección**   | Selecciona acciones que tienen un valor intrínseco superior al precio de mercado.      | Selecciona acciones de empresas con historial de fuerte crecimiento y potencial futuro. |
| **Tipo de Empresas Preferidas** | Tiende a favorecer empresas establecidas, con sólidos fundamentos y dividendos.       | Prefiere empresas con potencial de expansión futura.                              |
| **Métricas Clave**          | P/E bajo, P/B bajo, dividendos sólidos.                                               | Crecimiento de ingresos, ganancias, ROE, márgenes de beneficio.                  |
| **Enfoque en Dividendos**   | Sí, muchas acciones de valor pagan dividendos.                                        | No necesariamente; muchas empresas de crecimiento reinvierten ganancias en lugar de pagar dividendos. |
| **Horizonte de Inversión**  | A mediano plazo; busca beneficios a medida que el mercado reconoce el verdadero valor de las acciones. | A largo plazo; espera que el crecimiento de las empresas impulse el rendimiento.  |
| **Riesgos Considerados**    | Riesgo de valoración incorrecta; las acciones pueden permanecer subvaloradas durante períodos prolongados. | Riesgo de que las empresas de crecimiento no alcancen las expectativas de crecimiento; mayor volatilidad. |
| **Ejemplos de Inversores Famosos** | Warren Buffett, Benjamin Graham.                                                  | Peter Lynch.                                                                      |

    
La metodología que utiliza una visión "*en medio*" de ambas estrategias se denomina GARP (*Growth at Reasonable Price*).
    
###  Factores clave para seleccionar empresas usando Growth Investing  
    
    
1.- Crecimiento Histórico de las Ganancias.
       
    
2.- Márgenes de Beneficio Sólidos
    
    
3.- Fuerte Retorno sobre el Patrimonio 
    

4.- Fuerte Rendimiento de Acciones
    

A continuación, ahondaremos en cada punto y como descargar esta información utilizando yahoo finance

##  Práctica: Growth Investing  
    
    
### 0. Descarga de Fundamentales  

In [3]:
# Importar librerías
import numpy as np
import pandas as pd
import yfinance as yf
import datetime

import matplotlib.pyplot as plt

In [4]:
# Bajar un precio en una fecha dada
def get_historical_price(ticker: str, start: str):
    end = start+datetime.timedelta(days=4)
    price=pd.DataFrame(yf.Ticker(ticker).history(start=start, end=end)["Close"])
    return price.iloc[0, 0]

In [9]:
def get_financials(ticker:str):
    
    info = yf.Ticker(ticker)
    estado = info.income_stmt.iloc[:, :-1]
    balance = info.balance_sheet.iloc[:, :-1]

    return estado, balance

In [10]:
estado , balance = get_financials('LLY')

### 1.- Fuerte Crecimiento Histórico de Ganancias

Las empresas deben demostrar un historial de sólido crecimiento de ganancias en los últimos tres a diez años. El ratio financiero que nos puede ayudar a medir cuantitativamente esto es el EPS:
    
+ **EPS:** Múltiplio financiero que nos indica el beneficio por acción que tiene una compañía, en teoría, indica la cantidad de ingresos que le corresponde a cada inversionista por cada acción que tiene.
    
    
$$EPS = \frac{Income}{Shares}$$
    

    
    
El crecimiento mínimo de las ganancias por acción (EPS) en *Growth Investing* depende del tamaño de la empresa: por ejemplo, podrías buscar un crecimiento de al menos el 5% para empresas que superen los $4$ mil millones, el $7%$ para aquellas en el rango de $400$ millones a $4$ mil millones, y el $12%$ para empresas más pequeñas por debajo de los $400$ millones. La idea principal es que si la empresa ha mostrado un buen crecimiento en el pasado reciente, es probable que continúe haciéndolo en el futuro.

    
Además, las empresas objeto de inversión por medio de Growth Investing, muestran un price earnings ratio (*PER*) arriba de 25, lo que en Value Investing indicaría una sobre-valoración alta de un activo financiero.
    
    
+ **PER:** Indica cuántas veces se está pagando el beneficio neto anual de una empresa determinada al comprar una acción de esta, es decir, que tan cara o barata esta una acción, en función de su utilidad real y cuánto cuesta comprar una parte de la empresa.
    
$$PER = \frac{Price}{Income/Shares} = \frac{Price}{Earnings Per Share}$$


In [11]:
income = estado.loc['Net Income Common Stockholders']
no_shares = balance.loc['Ordinary Shares Number']


In [ ]:
Per = 

In [17]:
eps = income / no_shares
eps.pct_change(-1).fillna(0)

C:\Users\ppmel\AppData\Local\Temp\ipykernel_11908\2461831252.py:2: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  eps.pct_change(-1).fillna(0)


2024-12-31    1.024983
2023-12-31   -0.113436
2022-12-31    0.122886
2021-12-31    0.000000
dtype: float64

### 2.- Márgenes de Beneficio Sólidos  
    
    
El **profit margin** o margen de beneficio antes de impuestos de una empresa se calcula restando todos los gastos de las ventas (excepto impuestos) y dividiendo por las ventas. Es una métrica importante a considerar porque una empresa puede tener un crecimiento fantástico en ventas con ganancias pobres, lo que podría indicar que la administración no está controlando costos y ingresos.
    
 $$PM = \frac{\text{Income before Tax}}{\text{Sales or Gross Revenue}}$$

    
El porcentaje de profit margin considerado "bueno" depende de la industria, ya que los márgenes varían significativamente según el sector. Sin embargo, como regla general:
    
- **Financials:** 10%-20%    
    
    
- **Healthcare:**  15%-20%    
    
    
- **Consumer Staples:** 5%-10%   
    
 
- **Consumer Discretionary:** 10%-20%    
    

- **Technology:** 15%-25%  
    

- **Energy:** 5%-15%    
    

- **Real Estate :** 30%-40%    
    

- **Industrials:** 5%-10%    
    

- **Utilities:** 10%-15%    
    

- **Telecommunications:** 10%-15%    
    

- **Transportation:** 5%-10%  

### 3.- Fuerte Retorno sobre el Patrimonio  
    
El **ROE** de una empresa mide la rentabilidad al revelar cuánto beneficio genera con el dinero que los accionistas han invertido, en pocas palabras, que tanto provecho  le esta sacando una empresa al capital que  tiene:
    
$$ROE = \frac{Income}{Equity}$$
    
Se calcula dividiendo el ingreso neto entre el patrimonio de los accionistas. Una buena regla es comparar el ROE actual de una empresa con el ROE de tres-cinco años de la empresa y de la industria. Un ROE estable o en aumento indica que la administración está generando buenos rendimientos de las inversiones de los accionistas y opera eficientemente el negocio.
    
+ Si el resultado es inferior a 3, la situación es mala; si está entre 3 y 8, es normal; si es superior a 8, es buena. Se espera un resultado consistente y en crecimiento.

### 4.- Fuerte Rendimiento de Acciones  
    
    
Se buscan acciones que puedan que duplicar su crecimiento anual en cinco años, si no probablemente no sea una acción objeto de inversión en *growth investing*. Cabe destacar que al buscar un crecimiento tan fuerte, se debe de aceptar también una volatilidad mayor.
    
Ten en cuenta que el precio de una acción se duplicaría en siete años con una tasa de crecimiento del 11%. Para duplicarse en cinco años, la tasa de crecimiento debe ser de aproximadamente 15%, algo ciertamente factible para empresas en industrias de rápido crecimiento o con una ventaja competitiva.

